### ISSUE TITLE
Inferring protocol message value as function without ProtocolWithReturn

### ISSUE THREAD

**Issue by NoamLoewenstern** [aklinker1/webext-core#20]
Currently, if a message wants to get response, the way to do that is to use ProtocolWithReturn<data, responseValue>.
But it's possible to infer the type passing a regular function type, and inferring the data-argument and response-value from the type.

Instead of:
```Typescript 
interface ProtocolMap {
  getStringLength: ProtocolWithReturn<string, number>;
}
```
To infer like this:
```Typescript 
interface ProtocolMap {
  getStringLength: (str: string) => number;
  messageWithoutResponse: (num: number) => void;
  messageWithoutDataWithResponse: () => number;
}
```

**Comment by aklinker1** [comment]
Oh wow that's so much cleaner and easier to understand... Then I could support multiple arguments as well, and remove the weird internal typing.

Great recommendation! I'll see if I can support this without a breaking change, but I doubt that will be possible. I'll try and release v2 sometime today with this change.

**Comment by NoamLoewenstern** [comment]
I've added to [this pull request](https://github.com/aklinker1/webext-core/pull/21).
Seems there was an error in the pull-requests' bot deploying to vercel. Odd, since I've changed only a type check.

**Comment by aklinker1** [comment]
Thanks for the PR, I've got this merged and published to [`v1.3.0`](https://github.com/aklinker1/webext-core/releases/tag/messaging-v1.3.0). I also updated the docs: https://webext-core.aklinker1.io/messaging/protocol-maps.html
